projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -4592,4 +4855,134 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this.verticalScrollContainer.scrollPrevPage();
         this.nativeElement.focus();
     }
+
+    /**
+     * @hidden
+     */
+    public openRowEditingOverlay(row: IgxGridRowComponent) {
+        this.transactions.startPending();
+        this.configureRowEditingOverlay(row);
+        this.rowEditingOverlay.open(this.rowEditingOverlaySettings);
+        this.rowEditingOverlay.element.addEventListener('wheel', (evt) => { this.onRowEditingOverlayScroll(evt); });
+    }
+
+    /**
+     * @hidden
+     */
+    public repositionRowEditingOverlay(row: IgxGridRowComponent) {
+        this.configureRowEditingOverlay(row);
+        if (!this.rowEditingOverlay.collapsed) {
+            this.rowEditingOverlay.reposition();
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    public configureRowEditingOverlay(row: IgxGridRowComponent) {
+        if (this.rowEditingOverlaySettings) {
+            this.rowEditingOverlaySettings.positionStrategy.settings.target = row.element.nativeElement;
+            this.rowEditingOverlaySettings.outlet = this.rowEditingOutletDirective;
+            this.calculateRowChangesCount(row.rowID);
+            const lastIndex = this.rowList.length - 1;
+            const rowList = this.rowList.toArray();
+            if ((row.rowID === rowList[lastIndex].rowID ||
+                row.rowID === rowList[lastIndex - 1].rowID ||
+                row.rowID === rowList[lastIndex - 2].rowID)
+                // if row === rowList[0], then overlay should go to bottom, as otherwise it goes under the grid header
+                && row.rowID !== rowList[0].rowID) {
+                this.rowEditingOverlaySettings.positionStrategy.settings.verticalDirection = VerticalAlignment.Top;
+                this.rowEditingOverlaySettings.positionStrategy.settings.verticalStartPoint = VerticalAlignment.Top;
+            } else {
+                this.rowEditingOverlaySettings.positionStrategy.settings.verticalDirection = VerticalAlignment.Bottom;
+                this.rowEditingOverlaySettings.positionStrategy.settings.verticalStartPoint = VerticalAlignment.Bottom;
+            }
+
+            if (this.rowEditingOverlay.element.style.display === 'none') {
+                this.rowEditingOverlay.element.style.display = 'block';
+            }
+        }
+    }
+
+    public calculateRowChangesCount(rowID) {
```

Comment:
This one shold also be `@hidden`, right? Or not `public` at all if not needed for template bindings